modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v3_8_0/UpgradeDDMStructure.java

```diff
@@ -0,0 +1,399 @@
+		sb.append("join DDMStructureVersion on DDMStructure.structureid = ");
+		sb.append("DDMStructureVersion.structureid where ");
+		sb.append("DDMStructure.version = DDMStructureVersion.version and ");
+		sb.append("DDMStructure.classnameId = ?");
+
+		try (PreparedStatement ps1 = connection.prepareStatement(sb.toString());
+			PreparedStatement ps2 =
+				AutoBatchPreparedStatementUtil.concurrentAutoBatch(
+					connection,
+					"update DDMStructure set definition = ? where " +
+						"structureId = ?")) {
+
+			ps1.setLong(
+				1, PortalUtil.getClassNameId(DDMFormInstance.class.getName()));
+
+			try (ResultSet rs = ps1.executeQuery()) {
+				while (rs.next()) {
+					String structureVersionDefinition = rs.getString(
+						"definition");
+
+					ps2.setString(1, structureVersionDefinition);
+
+					long structureId = rs.getLong("structureId");
+
+					ps2.setLong(2, structureId);
+
+					ps2.addBatch();
+				}
+
+				ps2.executeBatch();
+			}
+		}
+	}
+
+	private void _upgradeDDMStructureLayout() throws Exception {
+		StringBundler sb = new StringBundler(11);
+
+		sb.append("select DDMStructureLayout.structureLayoutId, ");
+		sb.append("DDMStructureLayout.structureVersionId, ");
+		sb.append("DDMStructureLayout.definition as ");
+		sb.append("structureLayoutDefinition, DDMStructureVersion.definition ");
+		sb.append("as structureVersionDefinition from DDMStructureLayout ");
+		sb.append("inner join DDMStructureVersion on ");
+		sb.append("DDMStructureLayout.structureVersionId = ");
+		sb.append("DDMStructureVersion.structureVersionId inner join ");
+		sb.append("DDMStructure on DDMStructure.structureId = ");
+		sb.append("DDMStructureVersion.structureId where ");
+		sb.append("DDMStructure.classnameId = ?");
+
+		try (PreparedStatement ps1 = connection.prepareStatement(sb.toString());
+			PreparedStatement ps2 =
+				AutoBatchPreparedStatementUtil.concurrentAutoBatch(
+					connection,
+					"update DDMStructureLayout set definition = ? where " +
+						"structureLayoutId = ?")) {
+
+			ps1.setLong(
+				1, PortalUtil.getClassNameId(DDMFormInstance.class.getName()));
+
+			try (ResultSet rs = ps1.executeQuery()) {
+				while (rs.next()) {
+					String ddmStructureVersionDefinition = rs.getString(
+						"structureVersionDefinition");
+
+					String ddmStructureLayoutDefinition = rs.getString(
+						"structureLayoutDefinition");
+
+					long structureVersionId = rs.getLong("structureVersionId");
+
+					ps2.setString(
+						1,
+						_upgradeDDMStructureLayoutDefinition(
+							_getDDMForm(ddmStructureVersionDefinition),
+							ddmStructureLayoutDefinition, structureVersionId));
+
+					long structureLayoutId = rs.getLong("structureLayoutId");
+
+					ps2.setLong(2, structureLayoutId);
+
+					ps2.addBatch();
+				}
+
+				ps2.executeBatch();
+			}
+		}
+	}
+
+	private String _upgradeDDMStructureLayoutDefinition(
+		DDMForm ddmForm, String definition, long structureVersionId) {
+
+		List<Tuple> tuples = new ArrayList<>();
```

Comment:
@marcosapmf, variables of type `Tuple` aren't named only as `tuples` in Liferay code, usually, they're named as `entityTuples`, like `_serviceReferenceServiceTuples`. Can you please change it?

cc @carolmariaabb